int32[] flags
---
bool success
